contracts/moda-contracts/src/ModaRegistry.sol

```diff
@@ -3,22 +3,158 @@ pragma solidity 0.8.21;
 
 import "./interfaces/IModaRegistry.sol";
 import "./interfaces/IMembership.sol";
+import "@openzeppelin/contracts/access/extensions/AccessControlEnumerable.sol";
 
-contract ModaRegistry {
-    mapping(string => address) _catalogMemberships;
+contract ModaRegistry is IModaRegistry, AccessControlEnumerable {
+    using EnumerableSet for EnumerableSet.AddressSet;
 
+    /// State Variables
 
-    function isMember(string calldata catalogName, address user) external view returns (bool) {
-        address membership = _catalogMemberships[catalogName];
+    uint256 _catalogCount;
+    address payable _treasury;
+    uint256 _treasuryFee;
 
-        if (IMembership(membership).isMember(user)) {
+    mapping(string => Catalog) _catalogs;
+    mapping(address => EnumerableSet.AddressSet) _managers;
+
+    /// Errors
+
+    error CatalogAlreadyRegistered();
+    error CatalogDoesNotExist();
+    error NoCurrentCatalogs();
+
+    constructor() {
+        _grantRole(DEFAULT_ADMIN_ROLE, msg.sender);
+    }
+
+    /// Membership
+
+    /**
+     * @inheritdoc IModaRegistry
+     */
+    function isMember(string calldata name, address user) external view returns (bool) {
+        if (IMembership(_catalogs[name].membership).isMember(user)) {
```

Comment:
We can omit the `if` statements:
 
 ```sol
 return IMembership(_catalogs[name].membership).isMember(user);
 ```

Reply:
Let's remove `updateMerkleRoot` in `IMembership`. That should not be forced on anyone to implement as they may use another method to determine membership. I know it is not in this PR, but I came across it in VSCode

Reply:
Done